card/mod/carrierwave/lib/carrier_wave/file_card_uploader.rb

```diff
@@ -168,12 +172,13 @@ def filename
     end
 
     def extension
-      case
-      when file && file.extension.present? then ".#{file.extension}"
-      when card_content = model.content    then File.extname(card_content)
-      when orig = original_filename        then File.extname(orig)
-      else                                   ""
-      end.downcase
+      @extension ||=
```

Comment:
That looks dangerous to me. What if the file changes?

Reply:
I couldn't see how these objects could persist from one request to the next, so I didn't think there was much risk here.  However, the reward may not be big enough for much risk at atll, although I did see extension getting called a few different times and wondered whether we were repeating that processing.

Reply:
not request but what about file updates?

Reply:
```ruby
card.attachment.extension # => ".csv"
card.update_attributes! file: File.open("/Users/philipp/log.txt")
card.attachment.extension # => ".csv"
```

Reply:
without your caching the code above returns the correct updated extension